File: hunger-games-simulator/level/Tile.cs
﻿using hunger_games_simulator.lore.entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace hunger_games_simulator.level
{
    class Tile
    {
        public string AssetName;
        public char Char;
        public ConsoleColor Foreground, Background;
        public List<Entity> Entities;
    }
}

File: game/tiles/Tile.js

/**
 * @constructor
 */
function Tile(chr, foreground, background)
{
    var self = this;
    self.Char = chr;
    self.Foreground = foreground;
    self.Background = background;
    self.Entities = [];

    self.Asset = null;
}
Tile.RegisterTiles = function()
{
    Tile.Types = [];

    Tile.Types.push(LakeTile);
}